collect_app/src/main/java/org/odk/collect/android/utilities/DateTimeUtils.java

```diff
@@ -64,9 +66,12 @@ private static String getCustomDateTimeLabel(Date date, DatePickerDetails datePi
         } else if (datePickerDetails.isIslamicType()) {
             customDate = new DateTime(date).withChronology(IslamicChronology.getInstance());
             monthArray = context.getResources().getStringArray(R.array.islamic_months);
-        } else {
+        } else if (datePickerDetails.isNepaliType()) {
             customDate = new DateTime(date);
             monthArray = BsCalendar.MONTH_NAMES.toArray(new String[BsCalendar.MONTH_NAMES.size()]);
+        } else {
```

Comment:
I know this has been the approach for a while but we should have a better fallback. That is, this means that if you use the type `iislamic` you will end up with a Burmese calendar, right? How about having the else branch return the localized Gregorian label?

Reply:
This method is used only here: https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/utilities/DateTimeUtils.java#L37

That means if `date type` is different than Gregorian. If you use something strange like `iislamic` it's treated as Gregorian because doesn't match appearances we support and then this method is not called.

Reply:
I agree that this implicit `else` is fragile and confusing at a glance. Maybe just add a check for `isMyanmarType` here and add an `else` that logs an error?

Reply:
There is an issue https://github.com/opendatakit/collect/issues/2690 I'll take it once this one is merged so then we can come up with a better approach.

Reply:
How about using a switch case for all known date picker types and gregorian as default?